### PR TITLE
AC: Enable setting device_config to AC configs

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/config/config_reader.py [hunk 1]
```diff
@@ -419,6 +419,9 @@ def merge_modules(config, arguments, update_launcher_entry):
             if value:
                 update_launcher_entry['_{}'.format(key)] = value
 
+        if arguments_dict.get('device_config'):
+            update_launcher_entry['_device_config'] = read_yaml(arguments_dict['device_config'])
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/config/config_reader.py:423]: on my view, command line provided file should have priority under internal configuration or should be merged

File: tools/accuracy_checker/accuracy_checker/config/config_reader.py [hunk 2]
```diff
@@ -856,6 +859,11 @@ def _async_evaluation_args(launcher_entry):
         if not am.is_dir():
             launcher_entry['affinity_map'] = arguments.affinity_map
 
+    if 'device_config' in launcher_entry:
+        device_config = launcher_entry.pop('device_config')
+        if '_device_config' not in launcher_entry:
+            launcher_entry['_device_config'] = device_config
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/config/config_reader.py:865]: it is user provided argument, underscore used for private autogenerated, so ligic should be reversed for variables

Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/config/config_reader.py]: Done

File: tools/accuracy_checker/accuracy_checker/config/config_reader.py (hunk 1 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/config/config_reader.py]: Done

**Review by eaidova — APPROVED** [review]

**Pull request merged**